libglusterfs/src/strfd.c

```diff
@@ -15,6 +15,45 @@
 #include "glusterfs/strfd.h"
 #include "glusterfs/common-utils.h"
 
+#ifdef HAVE_OPEN_MEMSTREAM
+
+strfd_t *
+strfd_open(void)
```

Comment:
OK for stack allocation. BTW, I'm thinking about using strfd_t in glusterd_add_peers_to_auth_list() to replace bunch of error-prone calls to strlen() and str[n]cat().

Reply:
> OK for stack allocation. BTW, I'm thinking about using strfd_t in glusterd_add_peers_to_auth_list() to replace bunch of error-prone calls to strlen() and str[n]cat().

I think it's ok to use `open_memstream()` to replace strfd because the functionality is the same and the implementation inside glibc could be better/faster. However using it to replace some simple string functions (most of them using stack allocated buffers) could add unnecessary complexity and overhead.

However I agree that we should do something regarding the string manipulation in Gluster. I have some ideas that we can discuss if you like.

Reply:
> However I agree that we should do something regarding the string manipulation in Gluster. I have some ideas that we can discuss if you like.

Feel free to schedule Slack talk. I'm in Moscow, which is GMT+3, and usually available during regular working hours.